src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -269,8 +269,8 @@ public void HotReloadConfig()
                 DatabaseType.CosmosDB_NoSQL => HandleCosmosNoSqlConfiguration(graphQLSchema, runtimeConfig, connectionString),
                 _ => runtimeConfig with { DataSource = runtimeConfig.DataSource with { ConnectionString = connectionString } }
             };
-            ManagedIdentityAccessToken[_runtimeConfig.DefaultDataSourceName] = accessToken;
-            _runtimeConfig.UpdateDataSourceNameToDataSource(_runtimeConfig.DefaultDataSourceName, _runtimeConfig.DataSource);
+            ManagedIdentityAccessToken[_runtimeConfig.GetDefaultDataSourceName()] = accessToken;
+            _runtimeConfig.UpdateDataSourceNameToDataSource(_runtimeConfig.GetDefaultDataSourceName(), _runtimeConfig.DataSource);
```

Comment:
Doesn't `UpdateDefaultDataSourceName(...)` handle updating the runtimeconfig object's relevant internal data structures, rendering   `_runtimeConfig.UpdateDataSourceNameToDataSource(...)` redundant?

Reply:
This is used slightly differently, where a datasource name and a data source are provided and then this function will update that data source name to match the provided data source.

this function is only called from the initialization function that is used in a hosted scenario, and so will not be effected by the changes that I made, which will only happen when we are loading from an actual file.

Reply:
thanks for elaborating, understood now.